aiohttp/compression.py

```diff
@@ -0,0 +1,198 @@
+import enum
+import zlib
+
+from .http_exceptions import ContentEncodingError
+
+
+try:
+    import brotli
+except ImportError:  # pragma: nocover
+    brotli = None
+
+
+if brotli is None:
+    DEFAULT_ACCEPT_ENCODING = 'gzip, deflate'
+else:
+    DEFAULT_ACCEPT_ENCODING = 'gzip, deflate, br'
+
+
+class ContentCoding(enum.Enum):
+    # The content codings that we have support for.
+    #
+    # Additional registered codings are listed at:
+    # https://www.iana.org/assignments/http-parameters/http-parameters.xhtml#content-coding
+    deflate = 'deflate'
+    gzip = 'gzip'
+    identity = 'identity'
+    br = 'br'
+
+    @classmethod
+    def get_from_accept_encoding(cls, accept_encoding):
+        accept_encoding = accept_encoding.lower()
```

Comment:
Would be nice to parse `AcceptEncoding` header properly and respect weights instead of selecting the first supported compression based on **server** preference order.
Multiple `Accept-Encoding` headers are allowed by HTTP spec also.
We can either accept a list of headers (`req.headers.getall('Accept-Encoding')`) or `headers` param itself.